src/iop/channelmixer.c

```diff
@@ -99,6 +101,7 @@ typedef struct dt_iop_channelmixer_data_t
   float red[CHANNEL_SIZE];
   float green[CHANNEL_SIZE];
   float blue[CHANNEL_SIZE];
+  bool luminosity[CHANNEL_SIZE];
```

Comment:
Some time ago, the concerns about using bool were raised.
Better not use bool/boolean/gboolean in params.
Just use int.